std/experimental/allocator/package.d

```diff
@@ -443,6 +449,21 @@ auto make(T, Allocator, A...)(auto ref Allocator alloc, auto ref A args)
         return emplace!T(m, args);
     else return emplace(cast(T*) m.ptr, args);
 }
+/// ditto
+auto make(T, Allocator, P, A...)(auto ref Allocator alloc, auto ref P outer, auto ref A args)
+    if (is(T == class) && isInnerClass!T && is(P : AliasSeq!(__traits(parent, T))[0]))
```

Comment:
Could you hide this from the user? That's an implementation detail about which the user shouldn't be bothered (aka hide your specializations).

Moreover 
1) at #4680 I am trying to get `make` `@safe` under most circumstances, so this will expand the length of `make`
2) you create a lot of redundancies as only two lines are different

so I think it makes sense to stick with one method ;-)